charts/partials/templates/_topologyspreadconstraints.tpl

```diff
@@ -0,0 +1,23 @@
+{{ define "linkerd.topologySpreadConstraints" -}}
+{{- if .Values.enableTopologySpreadConstraints }}
+topologySpreadConstraints:
+- maxSkew: 1
+  topologyKey: failure-domain.beta.kubernetes.io/zone
+  whenUnsatisfiable: ScheduleAnyway
+  labelSelector:
+    matchExpressions:
+    - key: {{ default "linkerd.io/control-plane-component" .label }}
+      operator: In
+      values:
+      - {{ .component }}
```

Comment:
I think this can be simplified to this (same applies to rule below):
```suggestion
    matchLabels:
    - {{ default "linkerd.io/control-plane-component" .label }}: {{.component}}
```